k9mail/src/main/java/com/fsck/k9/view/RecipientSelectView.java

```diff
@@ -100,7 +105,90 @@ private void initView(Context context) {
         adapter = new RecipientAdapter(context);
         setAdapter(adapter);
 
-        setLongClickable(true);
+        addDragAndDropFunctionality(context);
+    }
+
+    public void setDragListener(DragListener listener) {
+        dragListener = listener;
+    }
+
+    private void addDragAndDropFunctionality(Context context) {
```

Comment:
this functionality deserves a class of its own, something like `RecipientDragAndDropUseCase`, that on creation receives the state it needs and a reference to the MvpView to do UI things, and exists as long as a drag is going on.